docs/docs/development/new-behavior.md

```diff
@@ -0,0 +1,383 @@
+                                                                            //                  Equal to n for behaviors that do make use of multiple instances)
+                      <behavior_name>_init, NULL,                           // Initialization Function, Power Management Device Pointer
+                      &<behavior_name>_data, &<behavior_name>_config,       // Behavior Data Pointer, Behavior Configuration Pointer (Both Optional)
+                      APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,     // Initialization Level, Device Priority
+                      &<behavior_name>_driver_api);                         // API Structure
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT) */
+
+```
+
+#### `DT_DRV_COMPAT`
+
+Replace `zmk_<behavior_name>` in the `#define DT_DRV_COMPAT` statement with the name of your behavior. (e.g. `zmk_behavior_caps_word`)
+
+#### Dependencies
+
+The dependencies required for any ZMK behavior are:
+
+- `device.h`: [Zephyr Device APIs](https://docs.zephyrproject.org/apidoc/latest/group__device__model.html)
+- `drivers/behavior.h`: ZMK Behavior Functions (e.g. [`locality`](#api-structure), `behavior_keymap_binding_pressed`, `behavior_keymap_binding_released`, `behavior_sensor_keymap_binding_triggered`)
+- `logging/log.h`: [Zephyr Logging APIs](https://docs.zephyrproject.org/latest/services/logging/index.html) (for more information on USB Logging in ZMK, see [USB Logging](usb-logging.md)).
+- `zmk/behavior.h`: ZMK Behavior Information (e.g. parameters, position and timestamp of events)
+  - `return` values:
+    - `ZMK_BEHAVIOR_OPAQUE`: Used to terminate `on_<behavior_name>_binding_pressed` and `on_<behavior_name>_binding_released` functions that accept `(struct zmk_behavior_binding *binding, struct zmk_behavior_binding_event event)` as parameters
+    - `ZMK_BEHAVIOR_TRANSPARENT`: Used in the `binding_pressed` and `binding_released` functions for the transparent (`&trans`) behavior
+  - `struct`s:
+    - `zmk_behavior_binding`: Stores the name of the behavior device (`char *behavior_dev`) as a `string` and up to two additional parameters (`uint32_t param1`, `uint32_t param2`)
+    - `zmk_behavior_binding_event`: Contains layer, position, and timestamp data for an active `zmk_behavior_binding`
+
+Other common dependencies include `zmk/keymap.h`, which allows behaviors to access layer information and extract behavior bindings from keymaps, and `zmk/event_manager.h`.
+
+##### ZMK Event Manager
+
+Including `zmk/event_manager.h` is required for the following dependencies to function properly.
+
+- `zmk/events/position_state_changed.h`: Position events' state (on/off), source, position, and timestamps
+- `zmk/events/keycode_state_changed.h`: Keycode events' state (on/off), usage page, keycode value, modifiers, and timestamps
+- `zmk/events/modifiers_state_changed.h`: Modifier events' state (on/off) and modifier value
+
+Events can be used similarly to hardware interrupts.
```

Comment:
I think it'd be great to include something here for subscriptions and listeners. Listeners are referred to below but they are not introduced. Maybe we could at least mention and add a link to Zephyr docs?

Reply:
Resolved in [aa17e24](https://github.com/zmkfirmware/zmk/commit/aa17e247bb23a6fefaf00be34aedf3ea376077ba)